airbyte-cdk/python/unit_tests/sources/file_based/scenarios/csv_scenarios.py

```diff
@@ -34,6 +35,158 @@
+                                },
+                                "globs": {
+                                    "title": "Globs",
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string"
+                                    }
+                                },
+                                "validation_policy": {
+                                    "title": "Validation Policy",
+                                    "type": "string"
+                                },
+                                "catalog_schema": {
+                                    "title": "ConfiguredAirbyteCatalog",
+                                    "type": "object",
+                                    "properties": {
+                                        "streams": {
+                                            "title": "Streams",
+                                            "type": "array",
+                                            "items": {
+                                                "$ref": "#/definitions/ConfiguredAirbyteStream"
+                                            }
+                                        }
+                                    },
+                                    "required": ["streams"]
+                                },
+                                "input_schema": {
+                                    "title": "Input Schema",
+                                    "type": "object"
+                                },
+                                "primary_key": {
+                                    "title": "Primary Key",
+                                    "anyOf": [
+                                        {
+                                            "type": "string"
+                                        },
+                                        {
+                                            "type": "array",
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        },
+                                        {
+                                            "type": "array",
+                                            "items": {
+                                                "type": "array",
+                                                "items": {
+                                                    "type": "string"
+                                                }
+                                            }
+                                        }
+                                    ]
+                                },
+                                "max_history_size": {
+                                    "title": "Max History Size",
+                                    "type": "integer"
```

Comment:
nit: I think this should have a default value. We might even want to hide the option until it needs to be customized since this is very much an advanced option

Reply:
When you mean hidden, do you mean hidden in that we don't even want it surfaced in the spec, or that we want to add `airbyte_hidden` so the UI knows not to include it?

The latter is doable, but just a bit ugly because we need to do some additional post processing (similar to us removing refs) to append it to the spec message that we return

Reply:
Is there a reason to include it in the spec if we're just going to hide it?

Reply:
I have a preference for removing the field from the config and letting connector developers configure it like Catherine proposed. We can add the field to the spec if/when there is a need for users to configure it

Reply:
okay sounds good, i can remove it entirely